druid/src/lens.rs

```diff
@@ -91,6 +91,22 @@ pub trait LensExt<A: ?Sized, B: ?Sized>: Lens<A, B> {
     {
         Then::new(self, other)
     }
+
+    /// Combine a `Lens<A, B>` with getter/setter functions to produce a `Lens<A, C>`
+    ///
+    /// ```
+    /// # use druid::*;
```

Comment:
Reading through these PRs, I am consistently wishing that there was more substantial documentation; ideally this would take the form of illustrating how this particular lens is useful in some hypothetical but realistic seeming scenario.

It's possible that this example could be that, but it would be nice to have a bit more hand holding, like "say you have data of type $type and a widget that expects a $something, using $this_lens, you can ...."

Reply:
basically: I think you're way down there in the lens mines working on your own, and those of us puttering around up here on the surface need a bit of help to understand the particular motivations for various designs. :)

Reply:
Added some more discussion.